### PR TITLE
Immutable upload file input

### PR DESCRIPTION
**Pull request by jrchudy**
This PR addresses issue #1253. The upload input was **NOT** being properly disabled in `recordedit` when the column had the immutable annotation defined. The input was also not having it's previous value showing properly in the input in this case.

### CODE REVIEW

**Comment by jrchudy** [comment]
@RFSH I added tests for disabled inputs in "edit" mode of `recordedit`. The spec I added those tests to is not the "best" place but does reduce the runtime of our tests by not creating a new spec. I can rename the spec to be `recordedit-disabled-inputs` instead of `recordadd-defaults`.

File: test/e2e/specs/default-config/recordedit/add-defaults.spec.js [hunk 1]
```diff
@@ -200,3 +216,48 @@ describe('Record Add with defaults', function() {
...
+            keys.push(testParams.edit_key.name + testParams.edit_key.operator + testParams.edit_key.value);
+            browser.ignoreSynchronization=true;
+            browser.get(browser.params.url + "/recordedit/#" + browser.params.catalogId + "/defaults:" + testParams.table_name + "/"  + keys.join("&"));
+
+            chaisePage.waitForElement(element(by.id("submit-record-button")));
+        });
+
+        for (var i=0; i < testParams.re_column_names.length; i++) {
+            (function(columnName) {
+                if (columnName == "foreign_key_disabled") {
```
Comment (RFSH) [inline test/e2e/specs/default-config/recordedit/add-defaults.spec.js]: I think using switch case is better than if else here.

Comment (jrchudy) [reply test/e2e/specs/default-config/recordedit/add-defaults.spec.js]: 👍

**Comment by jrchudy** [comment]
@RFSH I made the changes to rename the files and add the switch statement instead of the lengthy `if/else if/else` case. I will merge this once my other branch for the ACLs changes passes and is merged to master.

**Pull request merged**